sickbeard/postProcessor.py

```diff
@@ -1017,15 +1021,21 @@ def process(self):
         try:
             # move the episode and associated files to the show dir
             if self.process_method == "copy":
+                if helpers.isFileLocked(self.file_path, False):
+                    raise exceptions.PostProcessingFailed("File is locked for reading")
                 self._copy(self.file_path, dest_path, new_base_name, sickbeard.MOVE_ASSOCIATED_FILES,
                            sickbeard.USE_SUBTITLES and ep_obj.show.subtitles)
             elif self.process_method == "move":
+                if helpers.isFileLocked(self.file_path, True):
+                    raise exceptions.PostProcessingFailed("File is locked for reading/writing")
                 self._move(self.file_path, dest_path, new_base_name, sickbeard.MOVE_ASSOCIATED_FILES,
                            sickbeard.USE_SUBTITLES and ep_obj.show.subtitles)
             elif self.process_method == "hardlink":
```

Comment:
Nope, not here anyway.
If the destination file is locked you're doing something wrong, because the file shouldn't exist (and if it did, it gets removed elsewhere in the code). All this does is make sure that the source file is accessable for whatever operation you want to perform on it.
I'll add a proper check for write permissions on the destination later on. I also want to move all the file operations to one place, to make sure nothing can go wrong between deletion of the old files and placement of the new ones.

For now, this only checks the source file, and for hardlink you actually don't need access to the file itself. You can add a hardlink to a file that is completely inaccessable. Unfortunately that hardlink will become inaccessable itself, but the creation works. That's why I skipped the test for hardlinks. At least in windows.